Zend/zend.c

```diff
@@ -145,8 +145,13 @@ static ZEND_INI_MH(OnUpdateScriptEncoding) /* {{{ */
 static ZEND_INI_MH(OnUpdateAssertions) /* {{{ */
 {
 	zend_long *p = (zend_long *) ZEND_INI_GET_ADDR();
+	zend_string *errstr;
 
-	zend_long val = zend_atol(ZSTR_VAL(new_value), ZSTR_LEN(new_value));
+	zend_long val = zend_ini_parse_quantity(new_value, &errstr);
+	if (errstr) {
+		zend_error(E_WARNING, "Invalid \"%s\" setting: %s", ZSTR_VAL(entry->name), ZSTR_VAL(errstr));
```

Comment:
There are existing examples of E_WARNING in ini handlers wih the same error format, for example:

https://github.com/php/php-src/blob/2b1f8abd8778d14abeae6a83bc1205989f8bdece/ext/opcache/zend_accelerator_module.c#L210-L210